internal/play/play.go

```diff
@@ -38,11 +43,12 @@ func NewCommand() *cobra.Command {
 	}
 
 	cmd.Flags().String("device-id", "", "device ID from 'spotify device list'")
+	cmd.Flags().String("playlist", "", "playlist name from 'spotify playlist list'")
 
 	return cmd
 }
 
-func Play(api internal.APIInterface, query, deviceID string) (string, error) {
+func Play(api internal.APIInterface, query, contextQuery , deviceID string) (string, error) {
```

Comment:
I think we should pass `query` and `queryType` here, where `queryType` is "track", "album", or "playlist". Then we can `switch` on `queryType` to determine if we should do `internal.Search()` or a linear search through all of the user's playlists.

Reply:
An internal.Search() currently returns a *spotify.track not a *spotify.album/playlist
- So we need to modify that to allow for a return on an album. 
- If we do it for albums then the logic I have currently done for finding playlists can be overhauled aswel

Reply:
I suppose we should just get rid of `internal.Search()`, and instead call the API directly and return the PagingObject. Playing albums should have the same logic as playing tracks, just a different `queryType`. I like your current logic for playlists though. We should keep that.

Reply:
The logic for finding a playlist by name already exists for `spotify playlist show`, so we might be able to steal that code and use it here